libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp

```diff
@@ -768,6 +768,11 @@ bool WiFiClientSecure::_installClientX509Validator() {
 bool WiFiClientSecure::_connectSSL(const char* hostName) {
   _freeSSL();
   _oom_err = false;
+#ifdef DEBUG_ESP_SSL
+  if (!_use_insecure && !_use_fingerprint && !_use_self_signed && !_knownkey && !_sk) {
```

Comment:
What happens with this condition in the normal use case, i.e.: a certificate signed by an authority, which has been verified?

Reply:
It'll fail.  Needs to check that there is no CertStore and no CA list in the object as well.